marshal_test.go

```diff
@@ -2193,3 +2194,29 @@ func BenchmarkUnmarshalUUID(b *testing.B) {
 		}
 	}
 }
+
+func TestUnmarshalUDT(t *testing.T) {
+	info := UDTTypeInfo{
+		NativeType: NativeType{proto: 4, typ: TypeUDT},
+		Name:       "myudt",
+		KeySpace:   "myks",
+		Elements: []UDTField{
+			{
+				Name: "first",
+				Type: NativeType{proto: 4, typ: TypeAscii},
+			},
+			{
+				Name: "second",
+				Type: NativeType{proto: 4, typ: TypeSmallInt},
+			},
+		},
+	}
+	data := []byte("\x00\x00\x00\x0f\x00\x00\x00\x05Hello\x00\x00\x00\x02\x00\x2a")
```

Comment:
we should probably add a helper function to build these as they are bit opaque to glance at

Reply:
Do you mean something like the following?

```go
func bytesWithLength(data ...[]byte) []byte {
	totalLen := 0
	for i := range data {
		totalLen += len(data[i])
	}
	if totalLen > math.MaxUint32 {
		panic("total length overflows")
	}
	ret := make([]byte, totalLen+4)
	binary.BigEndian.PutUint32(ret[:4], uint32(totalLen))
	buf := ret[4:]
	for i := range data {
		n := copy(buf, data[i])
		buf = buf[n:]
	}
	return ret
}
```
```go
data := bytesWithLength( // UDT
    bytesWithLength([]byte("Hello")), // first
    bytesWithLength([]byte("\x00\x2a")), // second
)
```